packages/core/components/FileMetadataSearchBar/index.tsx

```diff
@@ -74,16 +95,26 @@ export default function FileMetadataSearchBar() {
         }
     }
 
-    function onDateRangeSelection(newDateRange: { startDate?: Date; endDate?: Date }) {
-        const [oldStartDate = undefined, oldEndDate = undefined] =
-            fileAttributeFilter?.value.split(DATE_RANGE_SEPARATOR) || [];
-        const startDate = newDateRange.startDate
-            ? newDateRange.startDate.toISOString().split("T")[0]
-            : oldStartDate;
-        const endDate = newDateRange.endDate
-            ? newDateRange.endDate.toISOString().split("T")[0]
-            : oldEndDate;
-        onSearch(`${startDate || endDate}${DATE_RANGE_SEPARATOR}${endDate || startDate}`);
+    function onDateRangeSelection(startDate: Date | null, endDate: Date | null) {
+        // Derive previous startDate/endDate from current filter state, if possible
+        let oldStartDate;
+        let oldEndDate;
+        const splitFileAttributeFilter = extractDatesFromRangeOperatorFilterString(
+            fileAttributeFilter?.value
+        );
+        if (splitFileAttributeFilter !== null) {
+            oldStartDate = splitFileAttributeFilter.startDate;
+            oldEndDate = splitFileAttributeFilter.endDate;
+        }
+
+        const newStartDate = startDate || oldStartDate || endDate;
+        const newEndDate = endDate || oldEndDate || startDate;
+        if (newStartDate && newEndDate) {
```

Comment:
When would this `if` statement not evaluate to `true`?

Reply:
It shouldn't ever in practice... but the linter kept yelling at me since they could technically both be `null` if I called the function with both parameters being `null`...